pinc/misc.inc

```diff
@@ -72,10 +77,12 @@ function array_get_as_array($array, $key, $default)
  * Remove fields from the associative array `$array` that aren't in the valid list.
  *
  * Note: This is an in-place operation that mutates `$array`!
+ *
+ * @param string[] $valid_fields
  **/
-function array_remove_invalid_fields(&$array, $valid_fields)
+function array_remove_invalid_fields(array &$array, ?array $valid_fields): void
 {
-    if ($valid_fields) {
+    if (!is_null($valid_fields)) {
```

Comment:
@70ray FYI this is a slight change in behavior as we will remove all keys if `$valid_fields` is set to `[]`. This seems reasonable to avoid returning unwanted keys but let me know what you think.

Reply:
Your change reflects the desired behavior, This makes sense to me.